### ISSUE TITLE
Tooltip on click should remain open until dismissed

### ISSUE THREAD

**Issue by petermakowski** [canonical-web-and-design/vanilla-framework#4412]
**Describe the bug**

-  Detached and regular Tooltip examples have a different behaviour.

- Both Detached[^1] and regular Tooltip should:
  - remain open until dismissed after opened by click (e.g. by using esc key or clicking outside).
  - close when pressing an `Esc` key

**Additional context**
Ideally both variants should behave in exactly the same way as it's just an implementation detail whether it's a detached tooltip or not.

Related issue in React components: https://github.com/canonical-web-and-design/react-components/issues/759

[^1]: https://vanillaframework.io/docs/patterns/tooltips#detached

**Comment by bartaz** [comment]
In Vanilla CSS framework the distinction between "detached" and standard tooltip is that the standard one is pure CSS, it is only triggered by hover and it can't be controlled with JavaScript (it can't stay open after mouse moves away, it can't be controled via key events). The "detached" variant is for situations when we want to control the tooltip via scripting (therefore this is the only "variant" that is controlled via JS and that's why it's used in react components implementation).

**Comment by petermakowski** [comment]
> In Vanilla CSS framework the distinction between "detached" and standard tooltip is that the standard one is pure CSS, it is only triggered by hover and it can't be controlled with JavaScript (it can't stay open after mouse moves away, it can't be controled via key events). The "detached" variant is for situations when we want to control the tooltip via scripting (therefore this is the only "variant" that is controlled via JS and that's why it's used in react components implementation).

Thanks! Gotcha, in that case let's limit the scope of this issue to the detached tooltip. Updated the description.

**Comment by bartaz** [comment]
@petermakowski Currently we have only examples that are opened by hover (even the detached one) - so I guess these should stay as they are (the show on hover, and are hidden when mouse moves away).

You want a separate behaviour for tooltips triggered by click? For them to behave like a contextual menu - stay open until dismissed?

**Comment by petermakowski** [comment]
> You want a separate behaviour for tooltips triggered by click? For them to behave like a contextual menu - stay open until dismissed?

That's the behaviour that I had in mind, yes. It seems that the regular tooltip example work in this way already - stays open until dismissed after click.

**Comment by petermakowski** [comment]
Just to clarify, what I mean (and what @rayito suggested) is that the tooltip should open on click as well as on hover/focus. With the only difference being that after you open a tooltip by click it only closes when you dismiss it (either by esc, or click outside).

**Comment by bartaz** [comment]
@bethcollins92 Can you try to add this functionality to our detached tooltip example?
